### PR TITLE
feat: Improved App Navigation

### LINKED ISSUE #19157: [Task]: Create Navigation Section Inside Settings Pane
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

https://www.notion.so/appsmith/Create-Navigation-Section-Inside-Settings-Pane-74636c80127e47fc8ac2c5d0c29f5004

Front logo [Front conversations](https://app.frontapp.com/open/top_7pnrg)

### LINKED ISSUE #19158: [Task]: Nav Component - Top + Stacked
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

https://www.notion.so/appsmith/Top-Stacked-97fc9945a35744e0b2575ea9d8070504

### LINKED ISSUE #19160: [Task]: Nav Component - Top + Inline
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

https://www.notion.so/appsmith/Top-Inline-b35e78ffdf0b4efb8335947f3a5872ed

### LINKED ISSUE #19161: [Task]: Live Preview Nav Configuration
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

https://www.notion.so/appsmith/Live-Preview-Nav-Configuration-31c67bcccbcc476fb51fd9fe2d5e6b24

### LINKED ISSUE #19173: [Task]: Nav Component - Sidebar

### LINKED ISSUE #19174: [Task]: Nav Component - Sidebar - Collapse
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

https://www.notion.so/appsmith/Sidebar-Collapse-69749b128aba4c75b1dfccaeb1521f8c

### LINKED ISSUE #20554: [Task]: Feedback from the DP - Improved App Nav
https://www.notion.so/appsmith/b7d76fc7f4cf4b49b879278ad697de7e?v=84e90d66c36e46d3ae1e5b6a6139a6cd&p=8bcd9453ec16484cb29cb97aac85ee77&pm=s

### LINKED ISSUE #20712: [Task]: Cypress / Jest - App Navigation
Test plan here - https://www.notion.so/appsmith/Test-Plan-a7883ae4980d470690de5c62a41dd168

### LINKED ISSUE #20938: [Task]: QA Callouts - Phase 1 - Improved App Nav
QA Issues here - https://docs.google.com/spreadsheets/d/1Kocq8h1H3EXlbqDgiNruzBr9MeNPyY26zct8IWYEY40/edit#gid=0

### LINKED ISSUE #21129: [Task]: QA Callouts - Phase 2 - Improved App Nav
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

QA Issues here - https://docs.google.com/spreadsheets/d/1Kocq8h1H3EXlbqDgiNruzBr9MeNPyY26zct8IWYEY40/edit#gid=0

### PR DESCRIPTION
**Pull request by dhruvikn**
## TL;DR
A new revamped experience for navigation for Appsmith users.

## Description 
Introduces new navigation styles with better default navigation - Top (Stacked), a variant for Top (Inline), and a collapsible Sidebar. Configure your app's navigation by navigating to the navigation settings tab inside the app settings pane and observe how your app with the selected navigation settings will look side by side as you change them.

This PR pushes the v1 for EPIC #17766.

Fixes #19157
Fixes #19158
Fixes #19174
Fixes #19173
Fixes #19160
Fixes #20712
Fixes #19161
Fixes #20554
Fixes #20938
Fixes #21129

## Media
<video src="https://user-images.githubusercontent.com/22471214/227187245-84e4e3fa-18e4-4690-8237-cfce29f432e5.mp4"></video>

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?
- Manual
- Cypress

### Test Plan
https://www.notion.so/appsmith/Test-Plan-a7883ae4980d470690de5c62a41dd168

### Issues raised during DP testing
https://docs.google.com/spreadsheets/d/1Kocq8h1H3EXlbqDgiNruzBr9MeNPyY26zct8IWYEY40/edit#gid=0

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/AppSettingsPane/AppSettings/NavigationSettings/LogoConfiguration.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import _ from "lodash";
+import React from "react";
+import { UpdateSetting } from ".";
+import StyledPropertyHelpLabel from "./StyledPropertyHelpLabel";
+
+const LogoConfiguration = (props: {
+  publishedNavigationSetting: PublishedNavigationSetting;
+  updateSetting: UpdateSetting;
+}) => {
+  const options = [
```
Comment (jsartisan) [inline app/client/src/pages/Editor/AppSettingsPane/AppSettings/NavigationSettings/LogoConfiguration.tsx:19]: Move this out of the functional component.

File: app/client/src/pages/Editor/AppSettingsPane/AppSettings/NavigationSettings/ColorStyleIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    backgroundColor =
+      selectedTheme?.properties?.colors?.primaryColor || Colors.WHITE;
+  } else if (props.colorStyle === NAVIGATION_SETTINGS.COLOR_STYLE.DARK) {
+    backgroundColor = Colors.GREY_900;
+  } else if (props.colorStyle === NAVIGATION_SETTINGS.COLOR_STYLE.LIGHT) {
+    backgroundColor = Colors.WHITE;
+  }
+
+  return (
+    <div
```
Comment (jsartisan) [inline app/client/src/pages/Editor/AppSettingsPane/AppSettings/NavigationSettings/ColorStyleIcon.tsx:27]: use styled.div for this

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]
